### PR TITLE
Distributed pkg

### LINKED ISSUE #50: Distributed package storage
Today the remote registry is a json file defined by config `command_repository_base_url`/index.json.

This index file contains the packages information:
```json
{
  name: "devenv",
  version: "1.0.48071",
  checksum: "xxx",
  startPartition: 0,
  endPartition: 9
}
```
The package itself is located at `command_repository_base_url`/<pkg_name>/<pkg_name>_<pkg_version>.pkg

Such setting requires all packages to release to the specific server.

By introducing a URL field into the package metadata, we can simplify the release process: calculating the checksum, and update the index.json file.

### LINKED ISSUE #53: Add --json option in config built-in command
Today the config command output is in plain text. The integration test uses grep to check if a field is present. If we support a json output format, it will allow us to make more advanced test.

### PR DESCRIPTION
**Pull request by bhou-crto**
close #50 
close #53

### CODE REVIEW

File: test/integration.sh [hunk 1]
```diff
@@ -147,6 +168,27 @@ else
   exit 1
 fi
 
+echo "> test update command updates bonjour package"
+RESULT=$($OUTPUT_DIR/cl)
+echo $RESULT | grep -q "bonjour"
+if [ $? -eq 0 ]; then
+  # ok
```
Comment (gmonceyron) [inline test/integration.sh:175]: Not sure this is necessary

File: examples/remote-config/remote_config.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-  "command_repository_base_url": "https://raw.githubusercontent.com/criteo/command-launcher/update-command/test/remote-repo",
+  "command_repository_base_url": "https://raw.githubusercontent.com/criteo/command-launcher/distributed-pkg/test/remote-repo",
```
Comment (gmonceyron) [inline examples/remote-config/remote_config.json:2]: Once this is merged, you will need another review to change the branch you are targeting for tests before deleting this branch, right ?

File: test/packages-src/bonjour/manifest.mf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (gmonceyron) [inline test/packages-src/bonjour/manifest.mf:1]: Is it worth keeping this in sources as it is also contained by the bonjour-v1.pkg file, and it is not used as a standalone file by tests ?

File: test/packages-src/bonjour/bonjour.bat [hunk 4]
```diff
@@ -0,0 +1 @@
+ECHO bonjour!
```
Comment (gmonceyron) [inline test/packages-src/bonjour/bonjour.bat:1]: Is it worth keeping this in sources as it is also contained by the bonjour-v1.pkg file, and it is not used as a standalone file by tests ?

File: test/integration.sh [hunk 5]
```diff
@@ -4,7 +4,7 @@ SCRIPT_DIR=${1:-$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && p
 echo "integration test directory: $SCRIPT_DIR"
 
 #BRANCH_NAME=$(git branch --show-current)
-EXAMPLE_BRANCH_NAME=main
+EXAMPLE_BRANCH_NAME=distributed-pkg
```
[the remark this answers is not in the document]
Comment (bhou-crto) [reply test/integration.sh:7]: Yes, you are right, the branch contains files that required for integration test. We don't need to change it soon, as long as we don't delete the branch. But I will  prepare a PR once it is merged

File: examples/remote-config/remote_config.json (hunk 2 above)
Comment (bhou-crto) [reply examples/remote-config/remote_config.json:2]: Yes

File: test/packages-src/bonjour/manifest.mf (hunk 3 above)
Comment (bhou-crto) [reply test/packages-src/bonjour/manifest.mf:1]: I think so based on 2 reasons:
1. with the source code, it make us easier to make further changes in the pkg. 
2. we can use the source as a dropin in the test as well.

File: test/integration.sh [hunk 6]
```diff
@@ -138,6 +158,7 @@ else
 fi
 
 echo "> test update command"
+# $OUTPUT_DIR/cl update --package
```
[the remark this answers is not in the document]
Comment (bhou-crto) [reply test/integration.sh]: Fixed

**Review by gmonceyron — APPROVED** [review]

**Pull request merged**